src/main/java/org/osc/controller/nsc/api/SampleSdnRedirectionApi.java

```diff
@@ -488,4 +494,124 @@ public void close() throws Exception {
         });
     }
 
+    // Support methods for REST API's
+
+    public List<String> getInspectionHooksIds() throws Exception {
+
+        return this.txControl.supports(() -> {
+            return this.utils.txInspectionHookEntities().stream().map(InspectionHookEntity::getHookId)
+                    .collect(Collectors.toList());
+        });
+    }
+
+    public Element updateInspectionPort(InspectionPortElement inspectionPort) throws Exception {
+
+        InspectionPortEntity inspectionPortEntity = (InspectionPortEntity) getInspectionPort(inspectionPort);
```

Comment:
lets do prevalidation of inspection port, ingrees/egress ports(both object and ids) before we get the entity out of the db

Reply:
@arvindn05, prevalidation of inspectionPort is already part of getInspectionPort method and hence it is not done.

Reply:
inspectionPort is already validated in getInspectionPort Method and for Ingress/Egress ports check is added to validate for id.